### PR TITLE
Add Integration Tests for Server Error

### PR DESCRIPTION
**Pull request by hazen**
- Clean up uncovered leaks in an error scenario
- Rename `riak_error` message to `riak_server_error` to avoid confusion
- Change how `riak_c_cunit` either runs only integration tests or unit tests

Note: This PR exercises error conditions so you will see these errors in the stderr output:

```
2014-03-23 00:50:29 EDT ERRO [27] ERR #0 - Invalid equality query undefined
2014-03-23 00:50:29 EDT CRIT [27] CALLBACK ERROR: Invalid equality query undefined
```

### CODE REVIEW

File: test/cunit/include/test_server_error.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*********************************************************************
+ *
+ * test_error.h: Riak C Unit testing for Server Errors
```
Comment (bookshelfdave) [inline test/cunit/include/test_server_error.h]: test_server_error.h

File: test/cunit/test_server_error.c [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    err = test_connect(cfg, &cxn);
+    CU_ASSERT_FATAL(err == ERIAK_OK)
+
+    test_bucket_key_value *db = NULL;
+    err = test_load_db(cfg, cxn, &db, 1, 1);
+    CU_ASSERT_EQUAL_FATAL(err,ERIAK_OK)
+
+    riak_2index_response *response;
+    err = riak_2index(cxn, db->bucket, db->key, NULL, &response);
+    CU_ASSERT_EQUAL_FATAL(err,ERIAK_SERVER_ERROR)
```
Comment (bookshelfdave) [inline test/cunit/test_server_error.c:57]: can you add a comment on why this would always fail?

**Comment by bookshelfdave** [comment]
```
Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      7      1    n/a      0        6
               tests    115     17     17      0        0
             asserts     91     91     91      0      n/a

Elapsed time =    0.166 seconds
```

what are the 6 inactive tests?

+1 otherwise

**Comment by hazen** [comment]
Dave - In this PR I changed how the integration test flag works:
- Without a command-line argument only the 6 unit test suites will run
- With a command-line argument only the 1 integration test suite will run

I did this so I can more easily do Valgrind on only the integration tests.  I have a task to go back and cleanup memory on the unit tests, which I'd do once the integration tests are all written.

**Comment by bookshelfdave** [comment]
right, I'm chalking it up to Sunday AM grogginess

**Pull request merged**